homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -88,11 +89,15 @@ async def async_setup_gateway_entry(
 
     async def async_update_data():
         """Fetch data from the subdevice."""
-        try:
-            for sub_device in gateway.gateway_device.devices.values():
+        data = {}
+        for sub_device in gateway.gateway_device.devices.values():
```

Comment:
Please extract the iteration and api call to a separate function that we can schedule once on the executor. Switching context is expensive.

Reply:
done